diff --git a/src/slicegpt/adapters/llama_adapter.py b/src/slicegpt/adapters/llama_adapter.py
index 1a7eb92..83aae85 100644
--- a/src/slicegpt/adapters/llama_adapter.py
+++ b/src/slicegpt/adapters/llama_adapter.py
@@ -224,7 +224,11 @@ class LlamaModelAdapter(ModelAdapter):
         local_files_only: bool = False,
         token: str | bool | None = None,
     ) -> ModelAdapter | None:
-        if not (model_name.startswith("meta-llama/Llama-2") or model_name.startswith("meta-llama/Meta-Llama-3")):
+        if not (
+            model_name.startswith("meta-llama/Llama-2")
+            or model_name.startswith("meta-llama/Meta-Llama-3")
+            or model_name.startswith("codellama/CodeLlama")
+        ):
             return None
 
         model = LlamaForCausalLM.from_pretrained(
@@ -244,7 +248,11 @@ class LlamaModelAdapter(ModelAdapter):
         local_files_only: bool = False,
         token: str | bool | None = None,
     ) -> ModelAdapter | None:
-        if not (model_name.startswith("meta-llama/Llama-2") or model_name.startswith("meta-llama/Meta-Llama-3")):
+        if not (
+            model_name.startswith("meta-llama/Llama-2")
+            or model_name.startswith("meta-llama/Meta-Llama-3")
+            or model_name.startswith("codellama/CodeLlama")
+        ):
             return None
 
         class UninitializedLlamaForCausalLM(LlamaForCausalLM):
